### PR TITLE
Shift `get_spike_times`  with `_t_start` when available

### PR DESCRIPTION
**Pull request by h-mayorquin**
As discussed with Alessio today this returns the right times when t start is available.

### CODE REVIEW

File: spikeinterface/core/basesorting.py [hunk 1]
```diff
@@ -58,16 +58,19 @@ def get_unit_spike_train(
...
+        spike_frames = segment.get_unit_spike_train(
             unit_id=unit_id, start_frame=start_frame, end_frame=end_frame).astype("int64")
         if return_times:
             if self.has_recording():
                 times = self.get_times(segment_index=segment_index)
-                return times[spike_train]
+                return times[spike_frames]
             else:
-                return spike_train / self.get_sampling_frequency()
+                segment = self._sorting_segments[segment_index]
```
Comment (alejoe91) [inline spikeinterface/core/basesorting.py:68]:
```suggestion
```
already defined above

Comment (h-mayorquin) [reply spikeinterface/core/basesorting.py:68]: True, it escaped me.

File: spikeinterface/extractors/neoextractors/neobaseextractor.py [hunk 2]
```diff
@@ -203,13 +203,13 @@ def _auto_guess_sampling_frequency(self):
 
 
 class NeoSortingSegment(BaseSortingSegment):
-    def __init__(self, neo_reader, segment_index, use_natural_unit_ids, t_start, sampling_freq):
+    def __init__(self, neo_reader, segment_index, use_natural_unit_ids, t_start, _sampling_frequency):
```
Comment (alejoe91) [inline spikeinterface/extractors/neoextractors/neobaseextractor.py:206]: why the `_`? It's an init argument, I'd make it just `sampling_frequency`

Comment (alejoe91) [inline spikeinterface/extractors/neoextractors/neobaseextractor.py:206]:
```suggestion
    def __init__(self, neo_reader, segment_index, use_natural_unit_ids, t_start, sampling_frequency):
```

File: spikeinterface/extractors/neoextractors/neobaseextractor.py [hunk 3]
```diff
@@ -203,13 +203,13 @@ def _auto_guess_sampling_frequency(self):
...
         BaseSortingSegment.__init__(self)
         self.neo_reader = neo_reader
         self.segment_index = segment_index
         self.use_natural_unit_ids = use_natural_unit_ids
         self._t_start = t_start
-        self._sampling_freq = sampling_freq
+        self._sampling_frequency = _sampling_frequency
```
Comment (alejoe91) [inline spikeinterface/extractors/neoextractors/neobaseextractor.py:212]:
```suggestion
        self._sampling_frequency = sampling_frequency
```

File: spikeinterface/extractors/neoextractors/neobaseextractor.py (hunk 2 above)
Comment (h-mayorquin) [reply spikeinterface/extractors/neoextractors/neobaseextractor.py:206]: Good catch, it was a mistake.

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**